Formula/opensearch-dashboards.rb

```diff
@@ -1,29 +1,56 @@
+require "language/node"
+
 class OpensearchDashboards < Formula
   desc "Open source visualization dashboards for OpenSearch"
   homepage "https://opensearch.org/docs/dashboards/index/"
   url "https://github.com/opensearch-project/OpenSearch-Dashboards.git",
-      tag:      "1.1.0",
-      revision: "44d2cb5b4f9a7c641c1fef32ec569bc48ec46979"
+      tag:      "1.2.0",
+      revision: "caf668e73304bac890f41c37cd6c3a41257cd289"
   license "Apache-2.0"
 
   bottle do
     sha256 cellar: :any_skip_relocation, all: "462dc37bc221e9c30afc971d931c7d0fdd50b66b42675e02056d140d49eaf1b3"
   end
 
-  depends_on "python@3.9" => :build
-  depends_on "yarn" => :build
+  depends_on "python@3.10" => :build
   depends_on "node@10" # Switch to `node` after https://github.com/opensearch-project/OpenSearch-Dashboards/issues/406
 
+  resource "yarn" do
+    url "https://registry.npmjs.org/yarn/-/yarn-1.22.17.tgz"
+    sha256 "267982c61119a055ba2b23d9cf90b02d3d16c202c03cb0c3a53b9633eae37249"
+  end
```

Comment:
Isn't this the same as the `yarn` formula?

Reply:
Oh, wait. That `yarn` uses `node`. Maybe it'll just work? 😄

Reply:
Ah, right, probably I should move up this comment from the bottom :)

>     # We can't depend on yarn formula since it brings the latest node
>     #  (which is getting picked up by some of the scripts and causes build failures)
>     # Repalce with `depends_on "yarn" => :build` when migraded to the latest node

Reply:
haha, found this, https://github.com/yarnpkg/website/issues/913#issuecomment-506660318

> arguing about this with Homebrew maintainers
> This has already been done countless times, all linked throughout this thread


Yarn can be installed without node, but underlyingly, yarn does need node though.

Reply:
Huh, interesting. I opened #90839, in case that's something we want to consider doing.

Reply:
yeah, looks good to me.

Reply:
You might also be interested about this nodejs TSC discussion, https://github.com/nodejs/TSC/issues/1012

Basically moving forward, nodejs would have corepack serve as abstraction to talk to nodejs package manager like `npm`, `yarn`, `pnpm`.

Reply:
Ok, I switched it back to `yarn` dependency. 
Also, it seems we don't need `python` to build the formula.